EventVisualisation/Workflow/src/EveWorkflowHelper.cxx

```diff
@@ -13,167 +13,205 @@
 /// \author julian.myrcha@cern.ch
 
 #include <EveWorkflow/EveWorkflowHelper.h>
+#include "ReconstructionDataFormats/GlobalTrackID.h"
+#include "EveWorkflow/FileProducer.h"
 #include "DataFormatsTRD/TrackTRD.h"
+
 #include "ITStracking/IOUtils.h"
 #include "DataFormatsGlobalTracking/RecoContainerCreateTracksVariadic.h"
+#include "DetectorsCommonDataFormats/NameConf.h"
+#include "DetectorsBase/Propagator.h"
 #include <type_traits>
 
 using namespace o2::event_visualisation;
 
-std::shared_ptr<const o2::event_visualisation::EveWorkflowHelper::tmpDataContainer>
-  o2::event_visualisation::EveWorkflowHelper::compute(const o2::globaltracking::RecoContainer& recoCont,
-                                                      const CalibObjectsConst* calib, GID::mask_t maskCl,
-                                                      GID::mask_t maskTrk, GID::mask_t maskMatch)
+void EveWorkflowHelper::selectTracks(const CalibObjectsConst* calib,
+                                     GID::mask_t maskCl, GID::mask_t maskTrk, GID::mask_t maskMatch)
 {
-  unsigned int nTracksTPCITSO2 = 0;
-  unsigned int nTRDTracksITSTPCTRD = 0;
-  unsigned int nTRDTracksTPCTRD = 0;
-  unsigned int nITSTPCTOFMatches = 0;
-  unsigned int nITSTPCTRDTOFMatches = 0;
-  unsigned int nTPCTRDTOFMatches = 0;
-  unsigned int nTPCTOFMatches = 0;
-
-  auto retVal = std::make_shared<tmpDataContainer>();
-
-  if (maskCl[GID::ITS]) {
-    const auto& ITSClusterROFRec = recoCont.getITSClustersROFRecords();
-    const auto& clusITS = recoCont.getITSClusters();
-    if (clusITS.size() && ITSClusterROFRec.size()) {
-      if (calib && calib->itsPatternDict) {
-        const auto& patterns = recoCont.getITSClustersPatterns();
-        auto pattIt = patterns.begin();
-        retVal->ITSClustersArray.reserve(clusITS.size());
-        o2::its::ioutils::convertCompactClusters(clusITS, pattIt, retVal->ITSClustersArray, *calib->itsPatternDict);
-      }
+  auto creator = [maskTrk, this](auto& trk, GID gid, float time, float) {
+    if (!maskTrk[gid.getSource()]) {
+      return true;
     }
+    mTrackSet.trackGID.push_back(gid);
+    mTrackSet.trackTime.push_back(time);
```

Comment:
note that for TPC tracks the time is in timeBins = 8 bunch-crossings = `8*o2::constants::lhc::LHCBunchSpacingMUS` in \mus

Reply:
ok - fixed for TPC (during drawing) - but for joined ITSTPC ... I assume time is in ms